prow/config/config.go

```diff
@@ -292,15 +292,26 @@ func loadConfig(prowConfig, jobConfig string) (*Config, error) {
 		return nil, fmt.Errorf("prowConfig cannot be a dir - %s", prowConfig)
 	}
 
-	nc, err := yamlToConfig(prowConfig, true)
-	if err != nil {
+	var nc Config
+	if err := yamlToConfig(prowConfig, &nc); err != nil {
+		return nil, err
+	}
+	if err := parseProwConfig(&nc); err != nil {
+		return nil, err
+	}
+	// Making a copy of job Config to merge it.
+	mainJobConfig := nc.JobConfig
+	// Resetting Job Config from the main config
+	nc.JobConfig = JobConfig{}
+
+	if err := nc.mergeJobConfig(mainJobConfig); err != nil {
```

Comment:
why?

Reply:
I did that when we re doing validation in mergeConfig such that we don't treat the main job config as a special case. Since we don't do validation anymore, we can probably undo it.